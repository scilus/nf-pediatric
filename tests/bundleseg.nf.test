nextflow_pipeline {

    options "-stub-run"

    name "Test nf-pediatric bundleseg"
    script "../main.nf"

    test("BundleSeg - should run successfully") {

        when {
            params {

                params.participant_label = ["sub-test1", "sub-test3", "sub-test4"]
                params.input_deriv = "$projectDir/tests/data/derivatives/"
                params.outdir = "$outputDir"

                params.bundling = true
            }
        }

        then {
            // stable name: All files + folders in ${params.outdir}/ with a stable name.
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore:  ['pipeline_info/*.{html,json,txt}'])
            assertAll {
                assert workflow.success
                assert snapshot(
                    // Number of successfully completed tasks
                    workflow.trace.succeeded().size(),
                    // Remove the nextflow version from the versions.yml because we test it using different nextflow versions.
                    removeNextflowVersion("$outputDir/pipeline_info/nf-pediatric_software_mqc_versions.yml"),
                    // All stable name.
                    stable_name
                ).match()
            }
        }
    }
}