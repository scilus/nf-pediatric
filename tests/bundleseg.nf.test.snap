{
    "BundleSeg - should run successfully": {
        "content": [
            33,
            {
                "BUNDLE_RECOGNIZE": {
                    "scilpy": "2.1.0"
                },
                "REGISTRATION_ANTS": {
                    "ants": "2.4.3",
                    "mrtrix": "3.0.4",
                    "imagemagick": null
                },
                "Workflow": {
                    "scilus/nf-pediatric": "v1.0dev"
                }
            },
            [
                "README.txt",
                "dataset_description.json",
                "multiqc",
                "multiqc/global_multiqc_data",
                "multiqc/global_multiqc_plots",
                "multiqc/global_multiqc_report.html",
                "pipeline_info",
                "pipeline_info/nf-pediatric_software_mqc_versions.yml",
                "sub-test1",
                "sub-test1/ses-baseline",
                "sub-test1/ses-baseline/dwi",
                "sub-test1/ses-baseline/dwi/bundles",
                "sub-test1/ses-baseline/dwi/bundles/sub-test1_ses-baseline_desc-AF_L.trk",
                "sub-test1/ses-baseline/dwi/bundles/sub-test1_ses-baseline_desc-AF_R.trk",
                "sub-test1/ses-baseline/dwi/sub-test1_ses-baseline_desc-endpoints_map_raw_stats.json",
                "sub-test1/ses-baseline/dwi/sub-test1_ses-baseline_desc-endpoints_metric_stats.json",
                "sub-test1/ses-baseline/dwi/sub-test1_ses-baseline_desc-length_stats.json",
                "sub-test1/ses-baseline/dwi/sub-test1_ses-baseline_desc-mean_std_per_point.json",
                "sub-test1/ses-baseline/dwi/sub-test1_ses-baseline_desc-mean_std_stats.json",
                "sub-test1/ses-baseline/dwi/sub-test1_ses-baseline_desc-streamline_count.json",
                "sub-test1/ses-baseline/dwi/sub-test1_ses-baseline_desc-volume_per_label.json",
                "sub-test1/ses-baseline/dwi/sub-test1_ses-baseline_desc-volume_stats.json",
                "sub-test1/ses-time1",
                "sub-test1/ses-time1/dwi",
                "sub-test1/ses-time1/dwi/bundles",
                "sub-test1/ses-time1/dwi/bundles/sub-test1_ses-time1_desc-AF_L.trk",
                "sub-test1/ses-time1/dwi/bundles/sub-test1_ses-time1_desc-AF_R.trk",
                "sub-test1/ses-time1/dwi/sub-test1_ses-time1_desc-endpoints_map_raw_stats.json",
                "sub-test1/ses-time1/dwi/sub-test1_ses-time1_desc-endpoints_metric_stats.json",
                "sub-test1/ses-time1/dwi/sub-test1_ses-time1_desc-length_stats.json",
                "sub-test1/ses-time1/dwi/sub-test1_ses-time1_desc-mean_std_per_point.json",
                "sub-test1/ses-time1/dwi/sub-test1_ses-time1_desc-mean_std_stats.json",
                "sub-test1/ses-time1/dwi/sub-test1_ses-time1_desc-streamline_count.json",
                "sub-test1/ses-time1/dwi/sub-test1_ses-time1_desc-volume_per_label.json",
                "sub-test1/ses-time1/dwi/sub-test1_ses-time1_desc-volume_stats.json",
                "sub-test3",
                "sub-test3/dwi",
                "sub-test3/dwi/bundles",
                "sub-test3/dwi/bundles/sub-test3_desc-AF_L.trk",
                "sub-test3/dwi/bundles/sub-test3_desc-AF_R.trk",
                "sub-test3/dwi/sub-test3_desc-endpoints_map_raw_stats.json",
                "sub-test3/dwi/sub-test3_desc-endpoints_metric_stats.json",
                "sub-test3/dwi/sub-test3_desc-length_stats.json",
                "sub-test3/dwi/sub-test3_desc-mean_std_per_point.json",
                "sub-test3/dwi/sub-test3_desc-mean_std_stats.json",
                "sub-test3/dwi/sub-test3_desc-streamline_count.json",
                "sub-test3/dwi/sub-test3_desc-volume_per_label.json",
                "sub-test3/dwi/sub-test3_desc-volume_stats.json",
                "sub-test4",
                "sub-test4/dwi",
                "sub-test4/dwi/bundles",
                "sub-test4/dwi/bundles/sub-test4_desc-AF_L.trk",
                "sub-test4/dwi/bundles/sub-test4_desc-AF_R.trk",
                "sub-test4/dwi/sub-test4_desc-endpoints_map_raw_stats.json",
                "sub-test4/dwi/sub-test4_desc-endpoints_metric_stats.json",
                "sub-test4/dwi/sub-test4_desc-length_stats.json",
                "sub-test4/dwi/sub-test4_desc-mean_std_per_point.json",
                "sub-test4/dwi/sub-test4_desc-mean_std_stats.json",
                "sub-test4/dwi/sub-test4_desc-streamline_count.json",
                "sub-test4/dwi/sub-test4_desc-volume_per_label.json",
                "sub-test4/dwi/sub-test4_desc-volume_stats.json"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-07-25T15:21:07.598622"
    }
}